src/main/java/io/cdap/plugin/SFTPCopyAction.java

```diff
@@ -206,9 +239,15 @@ private void copyJschZip(InputStream is, FileSystem fs, Path destination) throws
       while ((entry = zis.getNextEntry()) != null) {
         LOG.debug("Extracting {}", entry);
         Path destinationPath = fs.makeQualified(new Path(destination, entry.getName()));
-        try (OutputStream os = fs.create(destinationPath)) {
-          LOG.debug("Downloading {} to {}", entry.getName(), destinationPath.toString());
-          ByteStreams.copy(zis, os);
+        if (entry.isDirectory()) {
```

Comment:
is this possible for a zip?

Reply:
Yup, you can have embedded folders inside of a zip file. Original behavior was to fail upon encountering a folder in a zip file.